src/agora/doifft.cc

```diff
@@ -86,6 +88,32 @@ EventData DoIFFT::Launch(size_t tag) {
     }
   }
 
+  bool clipping = false;
+  float max_abs = 0;
+  for (size_t i = 0; i < 2 * cfg_->OfdmCaNum(); i++) {
+    float sample_val = ifft_out_ptr[i] / ifft_scale_factor_;
+    if (sample_val >= 1) {
+      clipping = true;
+      break;
+    }
+    if (std::abs(sample_val) > max_abs) max_abs = std::abs(sample_val);
+  }
+  if (clipping) {
+    std::stringstream ss;
+    ss << "WARNING: Clipping Occured in Frame " << frame_id << " Symbol "
+       << symbol_id << " Antenna " << ant_id << std::endl;
+    std::cout << ss.str();
+  }
+  if (ant_id < cfg_->BfAntNum() && max_abs < 1e-4) {
+    std::stringstream ss;
+    ss << "WARNING: Possibly bad antenna " << ant_id
+       << " with max sample value " << max_abs << std::endl;
+    std::cout << ss.str();
```

Comment:
Remove cout, and output using the agora logger.   AGORA_LOG_WARN or AGORA_LOG_ERR